clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,921 @@
+        show field keys       show field key information
+
+        A full list of influxql commands can be found at:
+        https://docs.influxdata.com/influxdb/latest/query_language/spec/`)
+	// ? Maybe add keybindings information like <CTRL+L> clearing the screen, <CTRL+D> exitting, etc
+}
+
+func (c *Client) settings() {
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 1, 1, ' ', 0)
+	fmt.Fprintln(w, "Setting\tValue")
+	fmt.Fprintln(w, "--------\t--------")
+	// TODO: No supported way to access or change host & port in a REPL session
+	// if c.Port > 0 {
+	// 	fmt.Fprintf(w, "Host\t%s:%d\n", c.Host, c.Port)
+	// } else {
+	// 	fmt.Fprintf(w, "Host\t%s\n", c.Host)
+	// }
+	fmt.Fprintf(w, "Username\t%s\n", c.Username)
+	fmt.Fprintf(w, "Database\t%s\n", c.Database)
+	fmt.Fprintf(w, "RetentionPolicy\t%s\n", c.RetentionPolicy)
+	fmt.Fprintf(w, "Pretty\t%v\n", c.Pretty)
+	fmt.Fprintf(w, "Format\t%s\n", c.Format)
+	fmt.Fprintf(w, "Precision\t%s\n", c.Precision)
+	fmt.Fprintln(w)
+	w.Flush()
+}
+
+func (c *Client) query(ctx context.Context, query string) (string, error) {
+	var resContentType string
+	switch c.Format {
+	case CsvFormat:
+		resContentType = "application/csv"
+	case JsonFormat, TableFormat:
+		resContentType = "application/json"
+	default:
+		return "", fmt.Errorf("unexpected format: %s", c.Format)
+	}
+	resBody, err := c.GetLegacyQuery(ctx).
+		U(c.Username).
+		P(c.Password).
+		Db(c.Database).
+		Q(query).
+		Rp(c.RetentionPolicy).
+		Accept(resContentType).
+		Execute()
+	if err != nil {
+		return "", err
+	}
+	return resBody, nil
+}
+
+func (c *Client) setFormat(args []string) {
+	// args[0] is "format"
+	if len(args) != 2 {
+		color.Red("Expected a format [csv, json, table]")
+		return
+	}
+	newFormat := FormatType(args[1])
+	switch newFormat {
+	case CsvFormat, JsonFormat, TableFormat:
+		c.Format = newFormat
+	default:
+		color.HiRed("Unimplemented format %q, keeping %s format.", newFormat, c.Format)
+		color.HiBlack("Choose a format from [csv, json, table]")
+	}
+}
+
+func (c *Client) setPrecision(args []string) {
+	// args[0] is "precision"
+	if len(args) != 2 {
+		color.Red("Expected a precision [ns, u, ms, s, m, or h]")
+		return
+	}
+	precision := args[1]
+	switch precision {
+	case "ns", "u", "µ", "ms", "s", "m", "h":
+		c.Precision = precision
+	default:
+		color.HiRed("Unimplemented precision %q, keeping %s precision.", precision, c.Precision)
+		color.HiBlack("Choose a precision from [ns, u, ms, s, m, or h]")
+	}
+}
+
+func (c *Client) outputCsv(csvBody string) {
+	fmt.Println(csvBody)
+}
+
+func (c *Client) outputJson(jsonBody string) {
+	if !c.Pretty {
+		fmt.Println(jsonBody)
+	} else {
+		var buf bytes.Buffer
+		if err := json.Indent(&buf, []byte(jsonBody), "", "  "); err != nil {
+			color.Red("Unable to prettify json response.")
+			fmt.Println(jsonBody)
+		} else {
+			fmt.Println(buf.String())
+		}
+	}
+}
+
+func (c *Client) outputTable(jsonBody string) {
```

Comment:
What happens if I redirect this to a file? Does the TUI do 'the right thing'?

Reply:
I think 'the right thing' is to do what e.g. `less` does with colors, and detect non-tty mode and switch to a simpler output with no colors. Maybe we can dependency-inject colors to be a no-op with no tty?